contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -1,87 +1,323 @@
+	// and cannot be changed.
 	LockToDefault bool
 	
-	// PreRelease indicates the maturity level of the feature. possible values are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", "featuregate.Deprecated"
+	// PreRelease indicates the maturity level of the feature. Possible values
+	// are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", or
+	// "featuregate.Deprecated"
 	PreRelease prerelease
+```
+
+Unless an exception is granted for a particular feature, as documented and
+approved as part of [Production Readiness Review], it is expected that:
+
+- Toggling a feature gate will not affect other components, e.g. disabling a
+  feature gate in the API server will work independently of disabling that same
+  feature gate in the kubelet or scheduler, and no coordination between
+  components is needed.
+- The effects of toggling a feature gate should be limited to the scope of the
+  feature. Enabling or disabling a feature gate should not affect workloads
+  that do not use the feature gate.
+- Toggling feature gates should not result in fanout effects or cascading
+  interactions in a cluster.
+- Disabling a feature gate should prevent any further damage caused by bugs in
+  the feature.
+
+[Production Readiness Review]: /sig-architecture/production-readiness.md
+
+## Enabling and disabling feature gates
+
+Gates can be enabled or disabled either via flags (see `--feature-gates`) or
+component config files (see `featureGates`).
+
+There is no supported way to change a feature gate at runtime for production
+Kubernetes use. A feature gate is typically toggled by a component restart.
+
+## Lifecycle
 
+Features generally progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end
+up deciding that a feature is not going to be supported and we end up marking
+them as `Deprecated`.
 
-## Alpha Features
+The majority of features will go through all three stages, but occasionally
+there are features which may skip a stage.
+
+### Alpha features
 
 * `PreRelease` is set to `featuregate.Alpha`
 * `Default` is always set to `false`
-* `LockToDefault` is not set. Defaults to `false`
+* `LockToDefault` is set to `false` (or not specified)
 
-By default Alpha features are not switched on. This enables folks to switch on the feature using the command line. All API changes must start with an Alpha gate, which makes it possible to rollback from future versions.
+Alpha features are never enabled by default, but users may switch them on
+explicitly.
 
-## Beta Features
+### Beta features
 
 * `PreRelease` is set to `featuregate.Beta`
 * `Default` is usually set to `true` (see below)
-* `LockToDefault` is not set. Defaults to `false`
+* `LockToDefault` is set to `false` (or not specified)
 
-Beta features are usually on by default. This enables the feature to be available in the default installation of Kubernetes.
+Beta features are usually enabled by default. This allows them to be available
```

Comment:
I found this a point of confusion also with kubernetes developers since beta APIs are disabled by default but the features can be enabled by default https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/3136-beta-apis-off-by-default , so beta features that need beta apis will not work by default :/